### ISSUE TITLE
Feature: Nushell support

### ISSUE THREAD

**Issue by charlesroper** [gerardog/gsudo#152]
It would be great if gsudo could support [Nushell](https://www.nushell.sh/).

It currently works fine if running a command like `gsudo notepad`. 

However, if I run just `gsudo` on its own to elevate the session, it starts an elevated `cmd` session rather than natively elevating Nushell.

If I run `gsudo nu` that works, and Starship picks up that I am running elevated. So hopefully a fix for this issue could be fairly easy?

![2022-08-23_03](https://user-images.githubusercontent.com/176/186260683-42cedbda-3227-4726-a937-2dc445778cc6.png)

**Comment by jmoore34** [comment]
A simple solution would be to put `alias su = sudo nu` in your config.nu

**Comment by gerardog** [comment]
I have a question... Actually two.

## 1. Shell elevation

When someone calls `sudo` or `gsudo`, it elevates the current shell.... From Nu that should elevate `nu.exe`. That is straightforward, except for the banner. There is no `nu -k` as in `cmd /k`, nor 'nu -NoLogo | -nol' as in pwsh, to skip the logo.
(Or is one? `--interactive`?)

![image](https://user-images.githubusercontent.com/3901474/187551983-bf69b4fe-d3f6-4696-9778-965c05224e2a.png)

## 2. command elevation
How gsudo should elevate commands when invoked from `nu`? Like in `gsudo {command}`.

This is different for every shell. 

For example In `cmd.exe`, shell elevation is done by elevating `cmd /k`, while command elevation is `cmd /c {command}`
- Should elevate as cmd commands? (as of right now) like `gsudo -d` would force from (for example) powershell or bash ?
- Should elevate as nu commands? as in: `gsudo ls /somedir"  | Take 1`. (it would become `<elevated> nu -c "ls" |  <non elevated>Take 1 `... and, what happens with the pipe, there? 
  - If we elevate the command between the pipes `| `, then we have two processes that need to communicate via Stdio/Stdout. And in nu we get different results when doing ` ls | Take 1` and `nu -c ls | Take 1`.
  - Could be a shell plugin or a profile function.: `sudo a|b|c` could elevate `a|b|c` if nu syntax allows it. 

Then, how to properly escape quoting...

**Comment by fdncred** [comment]
right now, there is no way to disable the banner from the command line. I thought this may work but it appears that the check for the banner config may happen in the wrong order to make this command useful. `nu -e 'let-env config = ($env.config | upsert show_banner false)'`

not sure about your second question

**Comment by charlesroper** [comment]
Thanks for looking at this @gerardog

I've set `show_banner: false` in my config, so that's not a problem for me. [See here](https://www.nushell.sh/blog/2022-08-16-nushell-0_67.html#themes-of-this-release).

I'm not quite sure about question 2 - I hadn't thought about it very deeply :)

If I run a command like `gsudo ls C:\Windows\ | Take 1` then I would expect the whole thing would run elevated as `nu` commands. I can put a `^` in front of a command to [escape to an external command](https://www.nushell.sh/book/escaping.html).

So `gsudo ls c:\Windows\System32 | take 1` would be like running `gsudo nu -c "ls c:\Windows\System32 | take 1"`.

Or `gsudo touch c:\Windows\System32\drivers\etc\hosts` would be like running `gsudo nu -c "touch c:\Windows\System32\drivers\etc\hosts"`

 It seems like an alias like @jmoore34 suggests could do it, but I'm not sure how to create an alias like this:

`alias sudo = (gsudo nu -c "$args")`

I've [asked on Discord](https://discord.com/channels/601130461678272522/601130461678272524/1015039577045610577). :)

**Comment by fdncred** [comment]
I posted this to discord.
```
def sudo [...rest] { gsudo $"nu -c '($rest | str collect ' ')'" }
```
but you still have to wrap the command in quotes like `sudo "ls | take 1"`. the things in quotes here are what shows up in the `$rest` variable.

The problem is that the nushell see's the pipe and stops collecting the rest because it thinks that's a new command which is why this doesn't work `sudo ls | take 1`

**Comment by charlesroper** [comment]
Thanks. That didn't work, but this slight modification does:

```nu
def sudo [...rest] { gsudo nu -c $"'($rest | str collect ' ')'" }
```

Shame it doesn't work without the quotes, but can live with this solution.

**Comment by fdncred** [comment]
Ya, that's what I meant. 🤣

**Comment by jmoore34** [comment]
> Thanks. That didn't work, but this slight modification does:
> 
> ```nu
> def sudo [...rest] { gsudo nu -c $"'($rest | str collect ' ')'" }
> ```
> 
> Shame it doesn't work without the quotes, but can live with this solution.

From what I can tell, $rest doesn't support passing over flags. E.g. I think you'll get an error from `sudo ls -l`

**Comment by charlesroper** [comment]
@jmoore34 Ah yes, good spot, you're right.

Your suggestion of a `su` alias works pretty well for my purposes actually. And just doing a `sudo nu -c "[nu commands]"` works fine too.

**Comment by gerardog** [comment]
Can a Nu function receive an anonymous method/block as a parameter? Like `sudo { ls -l }` ?
... if so, it could then escape the block content and call `gsudo nu -c "ls -l"`...

**Comment by gerardog** [comment]
Closing as implemented in v2.0.0
